### PR TITLE
Improve support for 32-bit EFI firmware in x86 ISOs and installer:

### PR DESCRIPTION
**Pull request by masato9000**
mklive
* include support for booting both 32- and 64-bit UEFI firmware with 32- and 64-bit images.
* require grub packages for both firmware bitness types (build-x86-images)

installer
* include support for installing both 32- and 64-bit UEFI grub loaders
* detect UEFI platform (firmware) bitness at install time

### CODE REVIEW

**Comment by masato9000** [comment]
@the-maldridge I was looking into this after seeing that the new x86_64 ISOs couldn't boot on 32-bit firmware, nor could i686 boot on 64-bit firmware.  (there are valid if uncommon reasons one might want to do the latter; at any rate, it's easier just to do it for both arches than for only one)

File: build-x86-images.sh.in [hunk 1]
```diff
@@ -25,10 +25,7 @@ readonly KDE_IMG=void-live-${ARCH}-${DATE}-kde.iso
 readonly LXDE_IMG=void-live-${ARCH}-${DATE}-lxde.iso
 readonly LXQT_IMG=void-live-${ARCH}-${DATE}-lxqt.iso
 
-case "$ARCH" in
-	i686*) GRUB="grub";;
-	x86_64*) GRUB="grub-x86_64-efi";;
-esac
+GRUB="grub-i386-efi grub-x86_64-efi"
```
Comment (the-maldridge) [inline build-x86-images.sh.in]: Since this doesn't need to be changed, perhaps it should be readonly.

File: mklive.sh.in [hunk 2]
```diff
@@ -209,41 +209,37 @@ generate_grub_efi_boot() {
     modprobe -q loop || :
 
     # Create EFI vfat image.
-    truncate -s 16M $GRUB_DIR/efiboot.img >/dev/null 2>&1
+    truncate -s 24M $GRUB_DIR/efiboot.img >/dev/null 2>&1
```
Comment (the-maldridge) [inline mklive.sh.in]: How much wiggle room does this give as the EFI stub may expand in size in the future?

File: build-x86-images.sh.in (hunk 1 above)
Comment (masato9000) [inline build-x86-images.sh.in]: if it's just nitpicks, I'm getting off easy ;)

`readonly GRUB=...` yeah, looking at the context, probably should be.  Done.

File: mklive.sh.in (hunk 2 above)
Comment (masato9000) [inline mklive.sh.in]: I'm not really sure.  Not a whole lot IIRC.  I looked at how much space was needed at the time I was messing with this, and gave it a little room, but I was leaning towards adding "just enough."

I remember you did 32 on yours, and I'd considered that, but I probably just got stuck in stone-age thinking (past dealings with embedded x86 systems where 8M actually made a difference between a working system and a nonworking system; but those were legacy systems anyway...)

so... Done.

**Comment by the-maldridge** [comment]
@voidlinux/pkg-committers I have tested these patches across the 4 hardware configurations that are affected and plan to merge on August 12th.  If you have breaking issues that will block this merge.  Please comment as such before the 12th of August.

**Comment by Gottox** [comment]
Go for it. I'd like to build new images and _test_ them afterwards.

**Comment by the-maldridge** [comment]
I personally prefer to test before merging, but that may be personal preference.

**Comment by Gottox** [comment]
Yea, I was pointing to the last release that included a non-functional X-Server on some systems.

**Pull request closed without merging**